weave-js/src/components/Tag/Tag.tsx

```diff
@@ -22,12 +23,18 @@ export const DEFAULT_TAG_ICON = 'tag';
 export function useTagClasses({
   color,
   isInteractive,
+  label,
 }: {
   color?: TagColorName;
   isInteractive?: boolean;
+  label?: string;
 }) {
-  const defaultColorRef = React.useRef(color ?? getRandomTagColor());
-  const tagColor = color ?? defaultColorRef.current;
+  const randomColor = useRef(color || label ? undefined : getRandomTagColor());
```

Comment:
```suggestion
  const randomColor = useRef((color ?? label) ? undefined : getRandomTagColor());
```

Reply:
Actually can we just make this block use conditionals? I think it might be worth it. Like: 

```javascript
function makeTagColor(color, label) {
   if (color) {
      return color
   }
   if (label) {
      return getTagColorByString(label)
   }
   return getRandomTagColor()
}
```

or something like that?

Reply:
or even
```javascript
const tagColor = color ? color : label ? getTagColorByString(label) : getRandomTagColor()
```

Reply:
The thing is we don't want to `useRef` for the color if it is assigned by the label. If you remove tags from the beginning of the list the colors stay the same by position, because of the `useRef`. We want the color to be a hash of the label and never vary by list order.

https://github.com/user-attachments/assets/74442a72-cdb9-44c1-a895-e8933f4248b5

Reply:
okay facts

Then maybe this? 

```javascript
const randomColor = useRef((color ?? label) ? undefined : getRandomTagColor());
const tagColor = color ? color : label ? getTagColorByString(label) : randomColor.current
````

Reply:
what you did was perfect i'm being a mega nit.